WordPress/Classes/ViewRelated/Notifications/Comment Moderation/CommentModerationView.swift

```diff
@@ -13,106 +13,182 @@ struct CommentModerationView: View {
             Divider()
                 .foregroundStyle(Color.DS.Background.secondary)
             VStack(spacing: .DS.Padding.double) {
-                titleHStack
-                mainActionView
-                secondaryActionView
+                switch viewModel.state {
+                case .pending:
+                    Pending(viewModel: viewModel)
+                case .approved(let liked):
+                    Approved(viewModel: viewModel, liked: liked)
+                case .trash, .spam:
+                    TrashSpam(viewModel: viewModel)
+                }
             }
             .padding(.horizontal, .DS.Padding.double)
         }
+        .animation(.smooth, value: viewModel.state)
     }
+}
+
+// MARK: - Subviews
+
+private struct Container<T: View, V: View>: View {
 
-    private var titleHStack: some View {
+    let title: String
+    let icon: V
+    let content: T
+
+    private let animationDuration = 0.25
+
+    private var animation: Animation {
+        return .smooth(duration: animationDuration)
+    }
+
+    init(title: String, @ViewBuilder icon: () -> V = { EmptyView() }, @ViewBuilder content: () -> T) {
+        self.content = content()
+        self.icon = icon()
+        self.title = title
+    }
+
+    var body: some View {
+        VStack(spacing: .DS.Padding.double) {
+            titleHStack
+            content
+        }
+        .padding(.horizontal, .DS.Padding.double)
+        .transition(
+            .asymmetric(
+                insertion: .opacity.animation(animation.delay(animationDuration * 0.8)),
```

Comment:
Nitpick: Should we not define a separate constant and specify that it is for insertion duration rather than multiplying it here?

Reply:
Resolved in fc2c17644f4e5ac1406b0ccb6204b1f5f1a92aa4